### DISCUSSION TITLE
Prioritization for Repositories

### DISCUSSION THREAD

**Discussion by jedgar2** [renovatebot/renovate#24845] [Suggest an Idea]
### What would you like Renovate to be able to do?
I know this is already implemented for packages, but would also be useful with regards to the order in which repositories are visited. This only makes sense within the context of autodiscover, where it would be an additional option to essentially start with certain repositories, then explore the others in any order. I have not tested the 'Repositories' array option which may have this effect if used in conjunction with autodiscover, but this is not well-documented if so.

### If you have any ideas on how this should be implemented, please tell us here.

.

### Is this a feature you are interested in implementing yourself?

Maybe

**Comment by viceice** [comment]
do you know auto discover filter?

https://docs.renovatebot.com/self-hosted-configuration/#autodiscoverfilter

**Comment by jedgar2** [comment]
Thanks for that. Helps with the exclusion, but would still find prioritization useful (unless that can be implemented by including the 'Repositories' array in conjunction with Autodiscover).

**Comment by viceice** [comment]
they are merged, but not sure about the order yet

**Comment by rarkins** [comment]
How would you anticipate this config working? e.g. what type of config file or CLI option would you use?

**Comment by viceice** [comment]
repositories and autodiscover are merged in autodiscover order (extras from repositories are removed)

**Comment by jedgar2** [comment]
> How would you anticipate this config working? e.g. what type of config file or CLI option would you use?

A few ideas would be either:

1) Allow the 'Autodiscover' property to be either a boolean, or an object/array containing the boolean, along with an ordered list of the first repositories that should be visited.
2) Have another property (e.g. autodiscoverPriority) with a similar structure (bar the boolean).
3) Modify the merging, such that the 'repositories' property is treated as a priority array if autodiscover is enabled, and make this clear in the docs.
4) ... (any other suggestions you may think suitable)

I feel option 2 is the best personally.

**Comment by viceice** [comment]
i think 2 is the way to go. it's pretty easy to implement.

**Comment by jedgar2** [comment]
I seen Philip left a comment on here wondering why we would want prioritization with regards to repositories since we "run over them all anyway" (looks like it has been deleted). Well the main use-case would be where a PR limit per run is set, in which case you won't be running over them all. Or alternatively, if you just have a very large codebase, it would enable you to get PRs out for those important repositories first.

**Comment by hoerup** [comment]
Some type of priority or a enqueue system would be very welcome.
We are running renovate onprem via helm against ~200 selfhosted gitlab repositories and a run takes on avg 15 minutes, so any method that could be used to prioritize a repo in front of the others will be very usefull.

Would it be possible to use redis cache to manage a queue of repos that should be scanned first?

Or if dashboard is enabled - maybe a checkbox where you can specify that this repo should be prioritized (this would of course need to be paired with a lookup of all dashboard tickets)
